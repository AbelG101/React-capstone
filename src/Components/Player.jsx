import { Link } from 'react-router-dom';

const Player = (props) => {
  const {
    name, age, nationality, img, clubName, appearences, minsPlayed, goals, assists,
  } = props;
  const state = {
    name, age, nationality, img, clubName, appearences, minsPlayed, goals, assists,
  };
  return (
    <Link
      to="/player-details"
      state={{
        state,
      }}
      className="player"
    >
      <div className="image"><img src={img} alt={name} /></div>
      <div className="info-container">
        <div className="info">
          <h5 className="name font-lato">{name}</h5>
          <h4 className="goals font-lato">
            {goals}
            {' '}
            goals
          </h4>
        </div>
        <i className="fa-solid fa-circle-arrow-right" />
      </div>
    </Link>
  );
};

export default Player;
