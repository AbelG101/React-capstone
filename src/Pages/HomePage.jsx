import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import Player from '../Components/Player';
import fetchTopScorers from '../Redux/reducer';

const HomePage = () => {
  const topScorers = useSelector((state) => state.reducer);
  const dispatch = useDispatch();

  useEffect(() => {
    if (topScorers.length === 0) {
      dispatch(fetchTopScorers());
    }
  });

  return (
    <>
      <header className="header">
        <h1>Premier League  ðŸ‡¬ðŸ‡§</h1>
        <h2>Top Scorers ðŸ¥…</h2>
      </header>
      <main className="home-container">
        {topScorers.map((scorer) => (
          <Player
            key={scorer.player.id}
            name={scorer.player.name}
            age={scorer.player.age}
            nationality={scorer.player.nationality}
            img={scorer.player.photo}
            clubName={scorer.statistics[0].team.name}
            appearences={scorer.statistics[0].games.appearences}
            minsPlayed={scorer.statistics[0].games.minutes}
            goals={scorer.statistics[0].goals.total}
            assists={scorer.statistics[0].goals.assists}
          />
        ))}
      </main>
    </>
  );
};

export default HomePage;
