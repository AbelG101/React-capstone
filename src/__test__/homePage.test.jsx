import renderer from 'react-test-renderer';
import { cleanup } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../configureStore';
import Player from '../Components/Player';
import HomePage from '../Pages/HomePage';

afterEach(cleanup);

test('Check if Home is rendering correctly', () => {
  const homePage = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <HomePage />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(homePage).toMatchSnapshot();
});

test('Check the player cards render correctly', () => {
  const state = {
    name: 'E. Haaland',
    age: 22,
    nationality: 'Norway',
    img: 'https://media.api-sports.io/football/players/278.png',
    clubName: 'Manchester City',
    appearences: 8,
    minutes: 665,
    total: 14,
    assists: 3,
  };
  const detailsPage = renderer
    .create(
      <Provider store={store}>
        <BrowserRouter>
          <Player
            name={state.name}
            age={state.age}
            nationality={state.nationality}
            img={state.photo}
            clubName={state.clubName}
            appearences={state.appearences}
            minsPlayed={state.minutes}
            goals={state.total}
            assists={state.assists}
          />
        </BrowserRouter>
      </Provider>,
    )
    .toJSON();
  expect(detailsPage).toMatchSnapshot();
});
