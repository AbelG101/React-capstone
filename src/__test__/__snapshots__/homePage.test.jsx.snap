// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Home is rendering correctly 1`] = `
Array [
  <header
    className="header"
  >
    <h1>
      Premier League  ðŸ‡¬ðŸ‡§
    </h1>
    <h2>
      Top Scorers ðŸ¥…
    </h2>
  </header>,
  <main
    className="home-container"
  />,
]
`;

exports[`Check the player cards render correctly 1`] = `
<a
  className="player"
  href="/player-details"
  onClick={[Function]}
>
  <div
    className="image"
  >
    <img
      alt="E. Haaland"
    />
  </div>
  <div
    className="info-container"
  >
    <div
      className="info"
    >
      <h5
        className="name font-lato"
      >
        E. Haaland
      </h5>
      <h4
        className="goals font-lato"
      >
        14
         
        goals
      </h4>
    </div>
    <i
      className="fa-solid fa-circle-arrow-right"
    />
  </div>
</a>
`;
